helpers/hjupyter.py

```diff
@@ -85,3 +95,214 @@ def build_run_notebook_cmd(
     ]
     cmd_run_txt = " ".join(cmd_run_txt)
     return cmd_run_txt
+
+
+# #############################################################################
+# NotebookImageExtractor
+# #############################################################################
+
+
+class NotebookImageExtractor:
+    """
+    Extract marked regions from a Jupyter notebook,
+    convert them to HTML and captures screenshots.
+
+    Initialize with input notebook path and output directory.
+    """
+
+    def __init__(self, notebook_path: str, output_dir: str) -> None:
+        self.notebook_path = notebook_path
+        self.output_dir = output_dir
+
+    def extract_regions_from_notebook(self) -> List[Tuple[str, str, List]]:
+        """
+        Extract regions from a notebook based on extraction markers.
+
+        This function reads a Jupyter notebook and searches for all regions
+        indicated by the markers inside cells:
+        ```
```

Comment:
I would move all the details in the markdown explaining the design and the implementation.